packages/transducers/src/iter/norm-range.ts

```diff
@@ -21,3 +21,37 @@ export function* normRange(
         }
     }
 }
+
+export function* normRange2d(
+    nx: number,
+    ny: number,
+    includeLastX: boolean = true,
+    includeLastY: boolean = true,
+) { 
+    const rx = normRange(nx, includeLastX);
```

Comment:
You're almost there - the only thing missing here is caching the result in an array: `const rx = [...normRange(nx, includeLastX)];`.

Reply:
Similar to the comment below, you could also express this using `mapcat`:

```ts
export function normRange2d(
    nx: number,
    ny: number,
    includeLastX: boolean = true,
    includeLastY: boolean = true,
) { 
    const rx = [...normRange(nx, includeLastX)];
    return mapcat((y) => map((x) => [x,y], rx), normRange(ny, includeLastY));
}
```

If you feel so inclined it'd be great to do some quick benchmarking for either approach. I've got a feeling the original approach will be slightly faster, but never know... :)